docs/developer-guide/variable-substitution.md

```diff
@@ -0,0 +1,119 @@
+---
+layout: page
+title: ksqlDB Variable Substitution
+tagline: Use variables in SQL statements
+description: Learn how to use variables in SQL statements
+---------------------------------------------------------
+
+Define variables
+----------------
+
+Syntax:
+```
+DEFINE <name> = '<value>';
+
+Where: 
+  <name>     is the variable name
+  <value>    is the variable value
+```
+
+Valid variables names start with a letter or underscore (\_) followed by zero or more alphanumeric characters or underscores (_).
+
+There are no type definition for values.  All variables values must be wrapped into single-quotes.
+
+Example:
+```
+DEFINE replicas = '3';
+DEFINE format = 'JSON';
+DEFINE name = 'Tom Sawyer';
+```
+
+Single-quotes are removed during variable substitution. If you need to escape single-quotes, then wrap the value with triple-quotes.
+
+Example:
+```
+# becomes 'my_topic'
+DEFINE topicName = '''my_topic''';      
+```
+
+Deleting variables
+------------------
+
+Syntax:
+```
+UNDEFINE name;
+```
+
+You can delete defined variables with the `UNDEFINE` syntax.
+
+Printing variables
+------------------
+
+Syntax:
+```
+SHOW VARIABLES;
+```
+
+Example:
+```
+ksql> DEFINE replicas = '3';
+ksql> DEFINE format = 'AVRO';
+ksql> DEFINE topicName = '''my_topic''';
+ksql> SHOW VARIABLES;
+
+ Variable Name | Value      
+----------------------------
+ replicas      | 3
+ format        | AVRO         
+ topicName     | 'my_topic' 
+----------------------------
+```
+
+Referencing substitution variables
+----------------------------------
+
+Variables are referenced by wrapping the variable name between `${}` characters (i.e. `${replicas}`).
+
+Example:
+```
+ksql> DEFINE format = 'AVRO';
```

Comment:
```suggestion
DEFINE format = 'AVRO';
```

Reply:
Usually, we don't include command prompts in example commands, to enable easy copy-paste from the page. But if there's example output, like you have in the previous example, we'll include the command prompts, for clarity.